sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/gcp.yaml

```diff
@@ -2,43 +2,33 @@ provider:
   metadata:
     name: GCP
     category: Cloud provider
-    references:
-      - description: OWASP Top 10 2021 Category A7 - Identification and Authentication Failures
-        link: https://owasp.org/Top10/A07_2021-Identification_and_Authentication_Failures/
-        type: Standards
-      - description: Google Cloud - Creating and managing service account keys
-        link: https://cloud.google.com/iam/docs/keys-create-delete
-        type: Documentation
-      - description: OWASP Top 10 2017 Category A3 - Sensitive Data Exposure
-        link: https://www.owasp.org/www-project-top-ten/2017/A3_2017-Sensitive_Data_Exposure
-        type: Standards
-      - description: MITRE, CWE-798 - Use of Hard-coded Credentials
-        link: https://cwe.mitre.org/data/definitions/798
-        type: Standards
-      - description: MITRE, CWE-259 - Use of Hard-coded Password
-        link: https://cwe.mitre.org/data/definitions/259
-        type: Standards
-      - description: SANS Top 25 - Porous Defenses
-        link: https://www.sans.org/top25-software-errors/#cat3
-        type: Standards
-    message: Make sure this GCP secret gets revoked, changed, and removed from the code.
+    message: Make sure this service account key gets revoked, changed, and removed from the code.
   detection:
     pre:
       include:
+        # GCP is not added there because of the risk of it being randomly
+        # included in a private key string, causing FPs.
         content:
-          - google
-          - gcp
+          - accounts.google.com
     post:
         patternNot:
           - "(\\w)\\1{6,}"
-          - "EXAMPLE"
-          - "PRIVATE_KEY"
-          - "REDACTED"
+          - "(?i)(?:s|ex)ample|redacted|private_key"
   rules:
     - rspecKey: S6335
       id: google-service-account-keys
       metadata:
         name: Google Service Account Keys
+      detection:
+        # Matches multiline private keys or single line private keys containing '\n's
+        matching:
+          pattern: "(\
+                    -----BEGIN PRIVATE KEY-----\
+                    (?:\\n|\\\\{1,2}n)\
```

Comment:
Maybe match also other whitespaces here? The key could be indented. You could also first match for the line break and only afterwards accept other whitespaces.
```suggestion
                    (?:\\s|\\\\{1,2}n)++\
```

Reply:
done !